### ISSUE TITLE
Sub annual behavior CAPACITY CONSTRAINT

### ISSUE THREAD

**Issue by DIUANA** [iiasa/message_ix#363]
Hi everybody I hope you are fine and safe.

I am working on a sub annual model and I have noticed some issues regarding the duration_time parameters and its impact on some equations in the model.

I have taken a look at some old issues such as 
https://github.com/iiasa/message_ix/issues/210
https://github.com/iiasa/message_ix/issues/191
https://github.com/iiasa/message_ix/issues/193

They are interesting but they do not adress my issue directly and even if they do there are no replies in two of them, maybe you have dealt with them internally but I believe you don't because it is still set as open.

Anyway talking about my issue, as far as I am concerned the equation CAPACITY_CONSTRAINT and MIN_UTILIZATION_CONSTRAINT (this one is being adressed on https://github.com/iiasa/message_ix/issues/371) might not be providing the expected output for sub annual models.

```
CAPACITY_CONSTRAINT(node,inv_tec,vintage,year,time)$( map_tec_time(node,inv_tec,year,time)
        AND map_tec_lifetime(node,inv_tec,vintage,year) )..
    SUM(mode$( map_tec_act(node,inv_tec,year,mode,time) ), ACT(node,inv_tec,vintage,year,mode,time) )
        =L= duration_time(time) * capacity_factor(node,inv_tec,vintage,year,time) * CAP(node,inv_tec,vintage,year) ;

```

CAPACITY CONSTAINT is active for every time since time set is among the equations variable sets, therefore I believe that include the duration_time parameter into the math may cause an issue. For instance, let's consider a model with two timeslices SUMMER and WINTER with a duration_time of 0.5 each one, a technology X with only mode 1 at node AUS in 2020. This technology must supply an annual demand of 10 hence its activity will be always 10 considering the relation ipt-otp = 1. Let's also assume it has a capacity facotr equals to 0.8.

**COMMODITY_BALANCE_GT(AUS,COM,LVL,2020,year)...  
         0.5 * ACT(AUS,X,2020,2020,1,SUMMER) + 0.5 * ACT(AUS,X,2020,2020,1,WINTER)
                   > = 10**

**CAPACITY_CONSTRAINT(AUS, X, 2020, 2020, SUMMER)....
                  ACT(AUS,X,2020,2020,1,SUMMER) = 10 <=
                                                                  0.5 (duration_time(SUMMER)) * 0.8(capacity_factor) * CAP**

In this case CAP must be at least 25 in order to make possible the technology X provides an ACT of 10, however it is not clear the duration_time should be considered because CAPACITY_CONSTRAINT and capacity_factor have been defined in the time level, the balance is not being applied for the year level in which duration_time must be considered, I believe the parameter should be **removed** or maybe replaced by _duration_time_rel_ like in the COMMODITY_BALANCE equation ( I did not think a lot about this option maybe it makes no-sense). 

I am wondering wheter my thoguths are toward the right direction. I have applied some tests  removing the duration_time from CAPACITY_CONSTRAINT equation:

```
CAPACITY_CONSTRAINT(node,inv_tec,vintage,year,time)$( map_tec_time(node,inv_tec,year,time)
        AND map_tec_lifetime(node,inv_tec,vintage,year) )..
    SUM(mode$( map_tec_act(node,inv_tec,year,mode,time) ), ACT(node,inv_tec,vintage,year,mode,time) )
        =L= capacity_factor(node,inv_tec,vintage,year,time) * CAP(node,inv_tec,vintage,year) ;
```

In that case, if we take the same example and apply the modified equations the outcome will be:

****CAPACITY_CONSTRAINT(AUS, X, 2020, 2020, SUMMER)....
                  ACT(AUS,X,2020,2020,1,SUMMER) = 10 <=
                                                                                  0.8(capacity_factor) * CAP**

In this case CAP would be bigger than 12.5 which I believe it makes more sense based on the ACT necessity, the capacity_factor and min_utilization_factor values.

In addition if we try to set capacity_factor equal to min_utilization_factor the modified equation works as good as the original one providing a CAP smaller than the old one. Let's assume min_utilization_factor = capacity_factor = 0.5. In that case based on the original equation CAP would be 40 and in the suggested one it would be 20 (considering the MIN_UTILIZATION_CONSTRAINT equation suggested in https://github.com/iiasa/message_ix/issues/371) which also makes more sense to supply a 10 ACT demand. 

Sorry for the long explanation.
I hope my idea is clear enough to make you able to get my point and explain to me if I am missing something.
All the Best
Fabio A. Diuana.

**Comment by behnam-zakeri** [comment]
@DIUANA thank you for openning this issue. As the first step to tacklae this, is it possible to answer this issue in two separate threads: one CAPACITY_CONSTRAINT and another one MIN_UTLIZIATION_CONSTRAINT? Or do you think they should be tackled together?

**Comment by DIUANA** [comment]
@behnam-zakeri I believe it is possible to addres them separetely but it is necessary to check if the results are fine and make sense. For example, the work around solution I applied has to be implemented together otherwise the results won't be good, for instance if cpacity_factor = min_utilization_factor both equations have to provide the same output value.

**Comment by behnam-zakeri** [comment]
@DIUANA , I suggest you open another issue for MIN_UTLIZATION_FACTOR, and we follow that case separately. I agree that at the end all these should ake sense together. But it's easier to quote-reply only related to one block of equations here. SO, I continue with your first question related to capacity factor here.

**Comment by DIUANA** [comment]
@behnam-zakeri I have done it now.

**Comment by behnam-zakeri** [comment]
This issue will be addressed through #509. Thanks a lot @DIUANA for reporting this, and please report if you have discovered additional issues.
